### ISSUE TITLE
Ruby Linux distribtests  are failing on master

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#18905]
https://source.cloud.google.com/results/invocations/408202cf-adc7-45ec-8d09-0045229efe70/targets/github%2Fgrpc/tests;group=tasks;test=distribtest.ruby_linux_x64_centos6_version_unspecified;row=1

I think very likely the offender is https://github.com/grpc/grpc/pull/18752 (it started failing after the PR was merged).

**Comment by apolcyn** [comment]
Hmm, I think that #18752 is triggering the failure, but the actual failure in the log is 

```
+ bundle exec ./distribtest.rb
bundler: failed to load command: ./distribtest.rb (./distribtest.rb)
LoadError: cannot load such file -- google/protobuf_c
  /usr/local/rvm/gems/ruby-2.2.10/gems/google-protobuf-3.7.1-x86_64-linux/lib/google/protobuf.rb:51:in `require'
  /usr/local/rvm/gems/ruby-2.2.10/gems/google-protobuf-3.7.1-x86_64-linux/lib/google/protobuf.rb:51:in `rescue in <top (required)>'
  /usr/local/rvm/gems/ruby-2.2.10/gems/google-protobuf-3.7.1-x86_64-linux/lib/google/protobuf.rb:48:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.2.10/gems/googleapis-common-protos-types-1.0.4/lib/google/rpc/status_pb.rb:4:in `require'
  /usr/local/rvm/gems/ruby-2.2.10/gems/googleapis-common-protos-types-1.0.4/lib/google/rpc/status_pb.rb:4:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/google_rpc_status_utils.rb:16:in `require'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/google_rpc_status_utils.rb:16:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/errors.rb:16:in `require_relative'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/errors.rb:16:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc.rb:17:in `require_relative'
  /usr/local/rvm/gems/ruby-2.2.10/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc.rb:17:in `<top (required)>'
  /var/local/git/grpc/test/distrib/ruby/distribtest.rb:17:in `require'
  /var/local/git/grpc/test/distrib/ruby/distribtest.rb:17:in `<top (required)>'
+ FAILED=true
+ '[' '' '!=' '' '
```

.... which looks like a protobuf issue.

My first guess is that this issue was present before that PR, but that PR caused protobuf to actually be required for the first time.

**Comment by jtattermusch** [comment]
Let's figure this one out before cutting the release branch.

**Comment by apolcyn** [comment]
On top of the load errors for `protobuf_c` there's also an error that looks like gRPC's libc usage might have jumped: 

```
Using distribtest 0.0.1 from source at `.`
/var/local/git/grpc/test/distrib/ruby/bin/distribtest.rb maybe `gem pristine distribtest` will fix it?
Bundle complete! 3 Gemfile dependencies, 7 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
+ bundle exec ./distribtest.rb
bundler: failed to load command: ./distribtest.rb (./distribtest.rb)
LoadError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.16' not found (required by /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/2.4/grpc_c.so) - /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/2.4/grpc_c.so
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/grpc.rb:20:in `require_relative'
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/grpc.rb:20:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/errors.rb:15:in `require_relative'
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc/errors.rb:15:in `<top (required)>'
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc.rb:17:in `require_relative'
  /usr/local/rvm/gems/ruby-2.4.1/gems/grpc-1.21.0.dev-x86_64-linux/src/ruby/lib/grpc.rb:17:in `<top (required)>'
  /var/local/git/grpc/test/distrib/ruby/distribtest.rb:17:in `require'
  /var/local/git/grpc/test/distrib/ruby/distribtest.rb:17:in `<top (required)>'
+ FAILED=true
+ '[' '' '!=' '' ']'
+ docker rm -f build_and_run_docker_7a3ca888-3757-4d82-90a7-8b7c60b7bc5b
build_and_run_docker_7a3ca888-3757-4d82-90a7-8b7c60b7bc5b
+ '[' true '!=' '' ']'
+ exit 1
2019-05-06 09:26:51,037 FAILED: distribtest.ruby_linux_x64_ubuntu1204_version_unspecified [ret=1, pid=27598, time=29.7sec]
```

**Comment by apolcyn** [comment]
I think there are a few different problems here:

1) #18875 caused gRPC's libc dependency when under binary compatibility mode to increase up to 2.16 - see [this comment](https://github.com/grpc/grpc/pull/18875#issuecomment-489724741) for details. This causes loader errors when using grpc-ruby on older platforms in the distrib tests.

2) #18752 is triggering a failure to load `protobuf_c.so` in all distrib tests that use ruby 2.2. The root cause is that protobuf dropped support for ruby 2.2 back in its 3.7 release, and the distrib test applications don't pin `google-protobuf`. However, until #18752, loading the pieces of grpc-ruby that are necessary to instantiate a channel didn't happen cause the `google-protobuf` library to also be loaded.

3) These distrib tests with the protobuf load errors are running ruby 2.2, which was EOL'd one year ago, and for which support has been already dropped by protobuf and cloud client libraries for example. It's probably time to follow suit and drop ruby 2.2 (https://github.com/grpc/grpc/issues/17604 for some previous conversation on that).

**Comment by apolcyn** [comment]
1) addressed in https://github.com/grpc/grpc/pull/18966
2) addressed in #18964

Planning to address 3) for the 1.22 branch

**Comment by apolcyn** [comment]
I'll track 3) in #18983.

This is fixed now, distrib tests are passing
